### PR TITLE
update (elmo): update elmo interface

### PR DESCRIPTION
**Pull request by kudep**
The elmo interface has been updated.
Short list of changes:
- the parameter "elmo_output_names" was added to use a combination of ELMo outputs
- added "concat_last_axis" if you want to concatenate all ELMo representations
- the parameter "max_token" is added - this is the limit of a number of words per line in a batch
- added the "mini_batch_size" parameter is used to reduce  memory requirements of a device

### CODE REVIEW

File: deeppavlov/models/embedders/elmo_embedder.py [hunk 1]
```diff
@@ -38,9 +38,11 @@ class ELMoEmbedder(Component):
     Parameters:
         spec: A ``ModuleSpec`` defining the Module to instantiate or a path where to load a ``ModuleSpec`` from via
             ``tenserflow_hub.load_module_spec`` by using `TensorFlow Hub <https://www.tensorflow.org/hub/overview>`__.
+        elmo_output_names: A list of output ELMo. You can use combination of ``["word_emb", "lstm_outputs1", "lstm_outputs2","elmo"]`` and you can use separately ``["default"]``. See `TensorFlow Hub <https://www.tensorflow.org/hub/overview>`__ for more information about it.
```
Comment (yoptar) [inline deeppavlov/models/embedders/elmo_embedder.py]: I would like to see a brief description for every name or a link to those descriptions as https://www.tensorflow.org/hub/overview has nothing of the sort.

Comment (kudep) [reply deeppavlov/models/embedders/elmo_embedder.py]: You're right, I changed the link to  a correct url.

File: deeppavlov/models/embedders/elmo_embedder.py [hunk 2]
```diff
@@ -129,40 +153,63 @@ def __call__(self, batch: List[List[str]],
...
+
+            if not self.concat_last_axis:
+                slice_indexes = np.cumsum(self.dim).tolist()[:-1]
+
+                def tok_f(vec):
+                    return np.array_split(vec, slice_indexes)
+
+                def batch_line_f(tokens):
+                    return list(map(tok_f, tokens))
+                elmo_output_values = list(map(batch_line_f, elmo_output_values))
```
Comment (yoptar) [inline deeppavlov/models/embedders/elmo_embedder.py]:
```
if not self.concat_last_axis:
    slice_indexes = np.cumsum(self.dim).tolist()[:-1]

    elmo_output_values = [[np.array_split(vec, slice_indexes) for vec in tokens]
                          for tokens in elmo_output_values]
```
?

File: deeppavlov/models/embedders/elmo_embedder.py [hunk 3]
```diff
@@ -129,40 +153,63 @@ def __call__(self, batch: List[List[str]],
...
+        if 'default' in self.elmo_output_names:
+            elmo_output_values = elmo_outputs['default']
+            dim0, dim1 = elmo_output_values.shape
             if self.dim != dim1:
-                batch = np.resize(batch, (dim0,self.dim))
+                shape = (dim0, self.dim if isinstance(self.dim, int) else self.dim[0])
+                elmo_output_values = np.resize(elmo_output_values, shape)
         else:
-            batch = elmo_outputs['elmo']
+            elmo_output_values = list(map(lambda elmo_output_name: elmo_outputs[elmo_output_name], self.elmo_output_names))
```
Comment (yoptar) [inline deeppavlov/models/embedders/elmo_embedder.py]: A list comprehension would look better here

File: deeppavlov/models/embedders/elmo_embedder.py [hunk 4]
```diff
@@ -12,18 +12,18 @@
...
 import tensorflow_hub as hub
-
 from deeppavlov.core.commands.utils import expand_path
-from deeppavlov.core.common.registry import register
-from deeppavlov.core.models.component import Component
 from deeppavlov.core.common.log import get_logger
+from deeppavlov.core.common.registry import register
 from deeppavlov.core.data.utils import zero_pad
+from deeppavlov.core.models.component import Component
+from overrides import overrides
```
Comment (yoptar) [inline deeppavlov/models/embedders/elmo_embedder.py]: This import should be placed in the middle bunch with numpy and tensorflow to comply with [pep8](https://www.python.org/dev/peps/pep-0008/?#imports)

File: deeppavlov/models/embedders/elmo_embedder.py (hunk 2 above)
Comment (kudep) [reply deeppavlov/models/embedders/elmo_embedder.py]: it is updated in 9efeba2 commit.

File: deeppavlov/models/embedders/elmo_embedder.py (hunk 3 above)
Comment (kudep) [reply deeppavlov/models/embedders/elmo_embedder.py]: it is updated in 9efeba2 commit.

File: deeppavlov/models/embedders/elmo_embedder.py (hunk 4 above)
Comment (kudep) [reply deeppavlov/models/embedders/elmo_embedder.py]: it is updated in 9efeba2 commit.

File: deeppavlov/models/embedders/elmo_embedder.py [hunk 5]
```diff
@@ -68,14 +75,44 @@ class ELMoEmbedder(Component):
 
 
     """
-    def __init__(self, spec: str, dim: int = 1024, pad_zero: bool = False, mean: bool = False,
-                **kwargs) -> None:
+    def __init__(self, spec: str, elmo_output_names: List = None, dim: int = None, pad_zero: bool = False,
```
[the remark this answers is not in the document]
Comment (kudep) [reply deeppavlov/models/embedders/elmo_embedder.py]: Changed. Look at ba92c52

**Pull request merged**